server/application-server/src/main/java/de/tum/in/www1/hephaestus/codereview/user/GHUser.java

```diff
@@ -0,0 +1,85 @@
+package de.tum.in.www1.hephaestus.codereview.user;
+
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.lang.NonNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import de.tum.in.www1.hephaestus.codereview.comment.IssueComment;
+import de.tum.in.www1.hephaestus.codereview.pullrequest.PullRequest;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.OneToMany;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+
+@Entity
+@Table(name = "gh_user")
+@Getter
+@Setter
+@NoArgsConstructor
+@ToString
+public class GHUser {
```

Comment:
Does it make sense to generalize it? GitLab has also roughly the same fields, same with "Repository" on GH and "Project" on GL.

We could also add our own user attributes here if necessary. I think it makes sense to keep it general

Reply:
So you would suggest a super-class (e.g. `ProviderUser`) and then sub-classes for GHUser and GLUser? Or just one class for all of them?
Also we might want to leave the `class User` open for when we create our own login/account system.

Reply:
Just use one class for them and call also call it user for now. GitHub and GitLab use a similar authentication, me might add the auth stuff to the same table since every user might also login with Hephaestus.